File: samos/sdk/js/quick_test.js

const { SamOSClient } = require('./index');

(async () => {
  const c = new SamOSClient({ baseUrl: 'http://localhost:8000' });
  const session = await c.startSession();
  const sid = session.session_id;
  console.log('Session:', session);

  console.log('Mode before:', await c.getMode(sid));
  console.log('Set mode to sandbox:', await c.setMode(sid, 'sandbox'));

  await c.putMemory(sid, 'demo.note', 'hello from node', { source: 'poc' });
  console.log('Memory get:', await c.getMemory(sid, 'demo.note'));
  console.log('Image:', await c.generateImage(sid, 'sunrise over calm sea'));
  console.log('EMMs:', await c.listEmms(sid, 5));
})().catch(e => { console.error(e); process.exit(1); });

File: samos/sdk/python/quick_test.py
from samos.client import SamOSClient

if __name__ == "__main__":
    c = SamOSClient(base_url="http://localhost:8000")
    session = c.start_session()
    sid = session["session_id"]
    print("Session:", session)

    print("Mode before:", c.get_mode(sid))
    print("Set mode to sandbox:", c.set_mode(sid, "sandbox"))

    c.put_memory(sid, "demo.note", "hello world", meta={"source": "poc"})
    print("Memory get:", c.get_memory(sid, "demo.note"))
    print("Image:", c.generate_image(sid, prompt="sunrise over calm sea"))
    print("EMMs:", c.list_emms(sid, limit=5))
